src/server/app.rs

```diff
@@ -41,28 +42,40 @@ pub async fn serve_archive(
                 "error: could not connect to database. Confirm that DATABASE_URL env var is set correctly."
             );
             tracing::error!("Error: {:?}", err);
-            process::exit(1);
+            return Err(CliError::DatabaseConnectionError);
         }
     };
 
-    let archive = Archive::parse(archive_path, &PathBuf::from(raw_archive_path), individual)
-        .unwrap_or_else(|err| {
+    let archive = match Archive::parse(archive_path, &PathBuf::from(raw_archive_path), individual) {
+        Ok(archive) => archive,
+        Err(err) => {
             tracing::error!("Unable to parse archive at '{raw_archive_path}'.");
-            tracing::error!("Error: {:?}", err);
-            process::exit(1);
-        });
+            tracing::error!("Error: {err:?}");
+            return Err(CliError::ArchiveParseError);
+        }
+    };
+
     let state = AppState { archive, db };
 
     HttpServer::new(move || {
         init_app(&state).unwrap_or_else(|err| {
             tracing::error!("Unable to initialize app.");
-            tracing::error!("Error: {:?}", err);
-            process::exit(1);
+            tracing::error!("Error: {err:?}");
+            // NOTE: We should not need to exit code 1 here (or in any of the closures in `routes.rs`).
```

Comment:
Interesting. I'm surprised the error here doesn't get propogated to the error in `run()` that your already catching. But I don't think panicing here is so bad, because it's just at app startup anyway, what do you think?

Reply:
Agree, I'll say we leave it here for now